pkg/manifest/file.go

```diff
@@ -50,6 +51,105 @@ type File struct {
 	readError error
 }
 
+// MarshalTOML performs custom marshalling to TOML for objects of File type.
+func (f *File) MarshalTOML() ([]byte, error) {
+	localServer := make(map[string]any)
+
+	if f.LocalServer.Backends != nil {
+		localServer["backends"] = f.LocalServer.Backends
+	}
+
+	if f.LocalServer.ConfigStores != nil {
+		localServer["config_stores"] = f.LocalServer.ConfigStores
+	}
+
+	if f.LocalServer.KVStores != nil {
+		kvStores := make(map[string]any)
+		for key, entry := range f.LocalServer.KVStores {
+			if entry.External != nil {
+				kvStores[key] = map[string]any{
+					"file":   entry.External.File,
+					"format": entry.External.Format,
+				}
+			} else {
+				items := make([]map[string]any, 0, len(entry.Array))
+				for _, e := range entry.Array {
+					obj := map[string]any{"key": e.Key}
+					if e.File != "" {
+						obj["file"] = e.File
+					}
+					if e.Data != "" {
+						obj["data"] = e.Data
+					}
+					items = append(items, obj)
+				}
+				kvStores[key] = items
+			}
+		}
+		localServer["kv_stores"] = kvStores
+	}
+
+	if f.LocalServer.SecretStores != nil {
+		secretStores := make(map[string]any)
+		for key, entry := range f.LocalServer.SecretStores {
+			if entry.External != nil {
+				secretStores[key] = map[string]any{
+					"file":   entry.External.File,
+					"format": entry.External.Format,
+				}
+			} else {
+				items := make([]map[string]any, 0, len(entry.Array))
+				for _, e := range entry.Array {
+					obj := map[string]any{"key": e.Key}
+					if e.File != "" {
+						obj["file"] = e.File
+					}
+					if e.Data != "" {
+						obj["data"] = e.Data
+					}
+					items = append(items, obj)
+				}
+				secretStores[key] = items
+			}
+		}
+		localServer["secret_stores"] = secretStores
```

Comment:
This is minor and non-blocking, but this loop block is nearly identical to the one above and could be refactored into a function used by both. Since we've had bugs in the past where behavior was changed for one type of store and not the other, reducing duplication will be beneficial.

Reply:
KV Store and Secret Store do not have the same structure anymore so I'm going to leave this alone for now, please look at my newest commit.